## Bayes Theorem to Predict Occurrences of A Word after Another Word
<p>A = probability of "small"</p>
<p>B = probability of "a"</p>
<p>A|B = probability of "small" after "a"</p>

In [1]:
import re

In [12]:
s = "A small dog jumped over the fence. The dog used its small paws to jump over the fence. The fence was no challenge. A small man had a llama, but the tall woman had a small alpaca."

In [13]:
s = re.sub(r'[^\w\s]','',s)
s = s.lower()
print(s)

a small dog jumped over the fence the dog used its small paws to jump over the fence the fence was no challenge a small man had a llama but the tall woman had a small alpaca


In [14]:
w = re.split("\s", s)
print(w)

['a', 'small', 'dog', 'jumped', 'over', 'the', 'fence', 'the', 'dog', 'used', 'its', 'small', 'paws', 'to', 'jump', 'over', 'the', 'fence', 'the', 'fence', 'was', 'no', 'challenge', 'a', 'small', 'man', 'had', 'a', 'llama', 'but', 'the', 'tall', 'woman', 'had', 'a', 'small', 'alpaca']


In [15]:
count = 0
for i in w:
    count += 1
print(count)    

37


In [21]:
A_counts = 0
for i in w:
    if "small" == i:
        A_counts += 1
print("A_counts = ",A_counts)

B_counts = 0
for i in w:
    if "a" == i:
        B_counts += 1
print("B_counts = ",B_counts)

A = A_counts/count
B = B_counts/count
print("A = ",A)
print("B = ",B)

A_counts =  4
B_counts =  4
A =  0.10810810810810811
B =  0.10810810810810811


In [23]:
BA_counts = 0
for i in range(len(w)):
    if w[i] == "a" and w[i+1] == "small":
        BA_counts += 1
print("BA_counts = ",BA_counts)
    
BA = BA_counts/A_counts
print("BA = ",BA)

BA_counts =  3
BA =  0.75


In [24]:
AB = (BA*A)/B
print("Naive Bayes = ",AB)

Naive Bayes =  0.75


## Notes
<p>This is more of a corpus linguistics issue, but I am trying to find the probability of "small", or any searched word, occurring after the indefinite determiner "a" in "A small dog jumped over the fence. The dog used its small paws to jump over the fence. The fence was no challenge. A small man had a llama, but the tall woman had a small alpaca." </p>

<p>A = probability of "small"</p>
<p>B = probability of "a"</p>
<p>A|B = probability of "small" after "a"</p>

<p>This part involves a little math using Bayes Theorem.</p>
<p>A|B = (B|A*A)/B</p>

<p>I am just playing around with probability and doing it by hand to analyze a corpus, and I am just trying to check whether I figured this properly as I check the probability of the occurrences of words after one another by hand doing good ol' fashion math instead of over relying on modules and libraries from R or Python just to ensure I grasp this concept.</p>
<p>Calculating this by hand, I get</p>
<p>count of "small" =  4</p>
<p>count of "a" =  4</p>
<p>A =  0.10810810810810811</p>
<p>B =  0.10810810810810811</p>
<p>count of "small" occurring after "a" = 3</p>
<p>BA =  0.75</p>
<p>probability of "small" occurring after "a" = 0.75</p>